### ISSUE TITLE
Fastify plugin does not propagate the trace scope when a request is sent with a post body

### ISSUE THREAD

**Issue by EricZaporzan** [DataDog/dd-trace-js#830]
**Describe the bug**

I'm using dd-trace to automatically instrument Fastify. I was noticing that the trace scope isn't propagating when a request is made with any type of POST body, which is pretty common. The result of this is that the spans resulting from the request (DB calls, external HTTP calls, etc) are disjointed and showing in different traces. 

Steps to reproduce: 

Run the simple server here: 

```
const tracer = require('dd-trace');

tracer.init();

const fastify = require('fastify');

const app = fastify({
  logger: true
});

// Declare a route
app.post('/', (req, res) => {
  console.log(tracer.scope().active());

  res.send({ hello: 'world' });
});


// Run the server!
app.listen(3000, (err, address) => {
  if (err) {
    throw err;
  }

  app.log.info(`server listening on ${address}`)
});
```

Send a simple POST request without a body, observe that the `console.log` emits a DatadogSpan object.

`curl -X POST http://localhost:3000`

Send a simple POST request with any type of body, observe that the `console.log` emits `null`. 

`curl -X POST http://localhost:3000 -H 'Content-Type: text/plain' -d a`

`curl -X POST http://localhost:3000 -H 'Content-Type: application/json' -d {\"a\":\"b\"}`

* **Operation system:**

MacOS 10.15.2 (also observed on Ubuntu) 

* **Node version:**

v10.17.0, v12.14.0

* **Tracer version:**

0.16.3

* **Agent version:**

`datadog/agent:6`

**Comment by EricZaporzan** [comment]
My current horrible workaround involves passing the context manually from middleware to each route and then activating it:

```
const tracer = require('dd-trace');

tracer.init();

const fastify = require('fastify');

const app = fastify({
  logger: true
});

app.use((req, res, next) => {
  req.activeScope = tracer.scope().active();

  next();
});

const handler = (req, res) => {
  console.log(tracer.scope().active());

  res.send({ hello: 'world' });
};

app.post('/', (req, res) => {
  return tracer.scope().activate(req.req.activeScope, () => handler(req, res));
});

// Run the server!
app.listen(3000, (err, address) => {
  if (err) {
    throw err;
  }

  app.log.info(`server listening on ${address}`)
});
```

This is obviously far from ideal, so I'm open to suggestions!

**Comment by rochdev** [comment]
While it's not ideal, it's very close to what the plugin should be doing for you, so I'd say it's a good temporary workaround. It's definitely a bug that it's not happening. I won't be able to fix this for the next release this week, but it should be able to make it to the next release after that.

**Comment by clsbartek** [comment]
@rochdev Is this issue fixed? I'm using fastify 3.22.0 and dd-trace 1.5.1. My post declaration:
```
 fastify.post<{
    Body: any;
    Response: any;
  }>('/test', function (_request: FastifyRequest, reply: FastifyWrappableReply) {
    console.log(process.env.DD_CONTEXT_PROPAGATION);
    console.log(tracer.scope().active());
    console.log(tracer.scope().active()?.context().toTraceId());
    reply.status(200).send();
  });
```

For post without body the result looks like:
```
[1] true
[1] DatadogSpan {...}
[1] 3780510505200846220
```
For post with body scope().active() returns null. TraceId is undefined:
```
[1] true
[1] null
[1] undefined
```

**Comment by rochdev** [comment]
@clsbartek It should be fixed but it's possible there are additional cases that were not covered. Can you provide a complete reproduction snippet?
